cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -89,24 +92,32 @@ public void getProfiles(HttpRequest request, HttpResponder responder,
   @Path("/profiles/{profile-name}")
   public void getProfile(HttpRequest request, HttpResponder responder,
                          @PathParam("namespace-id") String namespaceId,
-                         @PathParam("profile-name") String profileName) throws NotFoundException {
-    // TODO: implement the method and remove the mock data
-    if (new NamespaceId(namespaceId).equals(NamespaceId.SYSTEM)) {
-      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(SYSTEM_PROFILE));
-    } else {
-      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(USER_PROFILE));
-    }
+                         @PathParam("profile-name") String profileName) throws NotFoundException, IOException {
+    ProfileId profileId = new ProfileId(namespaceId, profileName);
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(profileStore.getProfile(profileId)));
   }
 
   /**
    * Write a profile in a namespace.
    */
   @PUT
   @Path("/profiles/{profile-name}")
-  public void writeProfile(HttpRequest request, HttpResponder responder,
-                           @PathParam("namespace-id") String namespaceId,
-                           @PathParam("profile-name") String profileName) throws BadRequestException {
-    // TODO: implement the method
+  public void writeProfile(
+    FullHttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+    @PathParam("profile-name") String profileName) throws BadRequestException, IOException, AlreadyExistsException {
+    ProfileCreateRequest profileCreateRequest;
+    try (Reader reader = new InputStreamReader(new ByteBufInputStream(request.content()), StandardCharsets.UTF_8)) {
+      // TODO: validate the profileCreateRequest, the provisoner should exist and the property should be correct
```

Comment:
Is there any JIRA associated with the TODO to make sure it won't get left behind?